### PR TITLE
Added method removeEventListener to WebSocket interface

### PR DESCRIPTION
**Pull request by hpherzog**
Hey Guys,

added a new method "removeEventListener" to the WebSocket interface, because I think it's common to have the counterpart to addEventListener.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ build
 coverage
 
 builderror.log
+
+*.iml
```
Comment (lpinca) [inline .gitignore]: Can you please remove these changes to `.gitignore`? Ideally these should go in your `.gitignore_global`.

File: lib/WebSocket.js [hunk 2]
```diff
@@ -481,6 +481,25 @@ WebSocket.prototype.addEventListener = function(method, listener) {
   }
 };
```
Comment (lpinca) [inline lib/WebSocket.js]: Please add JSDoc for this method.

File: lib/WebSocket.js [hunk 3]
```diff
@@ -481,6 +481,25 @@ WebSocket.prototype.addEventListener = function(method, listener) {
...
+WebSocket.prototype.removeEventListener = function(method, listener) {
+  var listeners = this.listeners(method);
+  for (var i = 0; i < listeners.length; i++) {
+    if (listeners[i]._listener === listener) {
+      this.removeListener(method, listeners[i]);
+    }
+  }
+};
+
+WebSocket.prototype.hasEventListener = function(method, listener) {
```
Comment (lpinca) [inline lib/WebSocket.js]: This is used only for testing right? If so can we remove it?
In the test we can use something like this:

```js
assert.strictEqual(ws.listeners().find((listener) => listener._listener === openListener), undefined);
```

File: test/WebSocket.test.js [hunk 4]
```diff
@@ -1616,6 +1616,24 @@ describe('WebSocket', function() {
       });
     });
 
+    it('should remove added event listener', function(done) {
+      this.timeout(4000);
```
Comment (lpinca) [inline test/WebSocket.test.js]: There is no need to increase the timeout please keep the default value by removing this line.

File: test/WebSocket.test.js [hunk 5]
```diff
@@ -1616,6 +1616,24 @@ describe('WebSocket', function() {
...
+        function messageListener() {}
+        ws.addEventListener('open', openListener);
+        ws.addEventListener('message', messageListener);
+        assert.equal(ws.hasEventListener('open', openListener), true);
+        assert.equal(ws.hasEventListener('message', messageListener), true);
+        ws.removeEventListener('open', openListener);
+        ws.removeEventListener('message', messageListener);
+        assert.equal(ws.hasEventListener('open', openListener), false);
+        assert.equal(ws.hasEventListener('message', messageListener), false);
+        done();
```
Comment (lpinca) [inline test/WebSocket.test.js]: Close the server before ending the test.

**Comment by lpinca** [comment]
Merged as 261a91075dca2bcb00cc8ce70ca2c7bb330bf3a9.
Thanks!

**Comment by maxcnunes** [comment]
@lpinca any plan publishing it?

**Comment by lpinca** [comment]
@maxcnunes we are getting closer. I don't have an ETA but version 2.0.0 is not too far.

**Pull request closed without merging**